### PR TITLE
Use :mpltype:`color` for color types

### PR DESCRIPTION
**Pull request by timhoffm**
Follow up to #27200.

### CODE REVIEW

**Comment by timhoffm** [comment]
I'm unclear why the doc build is failing. I`ve only added ``:mpltype:`color` ``. That this works in principle has been shown in #27200. Also my local doc build works. Likely, I'm overlooking something 😞.

**Comment by timhoffm** [comment]
Found the bug in the `:mpltype:` role (first commit). Usage in docstrings: second commit.

File: lib/matplotlib/quiver.py [hunk 1]
```diff
@@ -179,7 +179,7 @@
     Minimum length as a multiple of shaft width; if an arrow length
     is less than this, plot a dot (hexagon) of this diameter instead.
 
-color : color or color sequence, optional
+color : :mpltype:`color` or color sequence, optional
```
Comment (QuLogic) [inline lib/matplotlib/quiver.py:182]: Should `color sequence` here be `list of :mpltype:color`?

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -1073,7 +1073,7 @@ def hlines(self, y, xmin, xmax, colors=None, linestyles='solid',
             Respective beginning and end of each line. If scalars are
             provided, all lines will have the same length.
 
-        colors : color or list of colors, default: :rc:`lines.color`
+        colors : :mpltype:`color` or list of colors, default: :rc:`lines.color`
```
[the remark this answers is not in the document]
Comment (timhoffm) [reply lib/matplotlib/axes/_axes.py]: 1) The canoncial description is "list of [type]", i.e. should be "list of color". However, this is currently quite inconsistent. I've only changed it when changing the text anyway.

2) For "color or list of color" I've refrained from linking the second color if that would make a a line continuation necessary. But that's debatable of course.

File: lib/matplotlib/quiver.py (hunk 1 above)
Comment (timhoffm) [reply lib/matplotlib/quiver.py]: On a general note, I've used a couple of search/replace patterns. It is likely that I have not found all color type definitions with that. I think it's enough to start with the bulk here and fix others later if we find them.

**Comment by timhoffm** [comment]
Semi-OT: I don't think our GitHub auto-labeling (ref. #27633) makes much sense here. Can we fine-tune this. Or make deactivation possible (maybe through a "no-auto-label" label)? Or should we just live with it?

**Review by ksunden — APPROVED** [review]

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -1073,7 +1073,7 @@ def hlines(self, y, xmin, xmax, colors=None, linestyles='solid',
             Respective beginning and end of each line. If scalars are
             provided, all lines will have the same length.
 
-        colors : color or list of colors, default: :rc:`lines.color`
+        colors : :mpltype:`color` or list of colorjm , default: :rc:`lines.color`
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py:1076]: Typo.
```suggestion
        colors : :mpltype:`color` or list of color, default: :rc:`lines.color`
```

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
> Semi-OT: I don't think our GitHub auto-labeling (ref. #27633) makes much sense here. Can we fine-tune this. Or make deactivation possible (maybe through a "no-auto-label" label)? Or should we just live with it?

It only cares about file names, so you can't have it ignore PRs where just docstring are touched, for example.

**Pull request merged**